js/test/units/testunitfmt_hy_AM.js

```diff
@@ -0,0 +1,331 @@
+    testUnitFormatArea2_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "square centimeter",
+            amount: 2
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM", 
+            autoConvert: true,
+            length: "long"
+        });
+
+        var str = uf.format(m1);
+        test.equal(str, "2 քառակուսի սանտիմետր");
+        test.done();
+    },
+    testUnitFormatArea3_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "hectare",
+            amount: 1000
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM", 
+            autoConvert: true,
+            length: "short"
+        });
+
+        var str = uf.format(m1);
+        test.equal(str, "10 կմ²");
+        test.done();
+    },
+    testUnitFormatArea4_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "hectare",
+            amount: 1000
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM", 
+            autoConvert: true,
+            length: "long"
+        });
+
+        var str = uf.format(m1);
+        test.equal(str, "10 քառակուսի կիլոմետր");
+        test.done();
+    },
+    testUnitFormatFuelConsumption1_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "km/liter",
+            amount: 2
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM",
+            autoConvert: true,
+            length: "short"
+        });
+        var str = uf.format(m1);
+        test.equal(str, "2 կմ/լ");
+        test.done();
+    },
+    testUnitFormatFuelConsumption2_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "km/liter",
+            amount: 2
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM",
+            autoConvert: true,
+            length: "long"
+        });
+        var str = uf.format(m1);
+        test.equal(str, "2 կիլոմետր լիտրի վրա");
+        test.done();
+    },
+    testUnitFormatLength1_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "mile",
+            amount: 2000
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM",
+            autoConvert: true,
+            length: "short"
+        });
+        var str = uf.format(m1);
+        test.equal(str, "3,21868 Mm");
+        test.done();
+    },
+    testUnitFormatLength2_hy_AM: function(test) {
+        test.expect(1);
+        var m1 = MeasurementFactory({
+            unit: "mile",
+            amount: 2000
+        });
+
+        var uf = new UnitFmt({
+            locale: "hy-AM",
+            autoConvert: true,
+            length: "long"
+        });
+        var str = uf.format(m1);
+        test.equal(str, "3,21868 megameters");
```

Comment:
"Megameters" in English?

Reply:
There was no translation data in hy/sysres.json. that's why it returns english. so I added it by using google translate. the default system is `metric` in hy-AM. so I added an additional test case with `meter`.

Reply:
Maybe just not test that particular case if it is not in CLDR. Only test the meters, or other case which is in CLDR.

Reply:
I updated it